config/routes.rb

```diff
@@ -12,4 +12,13 @@
       resources :likes, only:[:create, :destroy]
     end
   end
+  namespace :api do
+    namespace :v1 do
+      resources :users, only: [:index] do
+        resources :posts, only: [:index] do
+          resources :comments, only:[:index, :create]
+        end
+      end
+    end
+  end
 end
```

Comment:
Hey there! 👋

I noticed that the JSON API endpoints weren't working as expected. Here's what I found:

The endpoints, such as `/api/v1/users/:user_id/posts`, were not functioning because the routes were missing the `show` actions for both `users` and `posts`. 

To fix this, Here is a suggestion: you can update your routes and controllers like this:

```ruby
namespace :api do
  namespace :v1 do
    resources :users, only: [:index, :show] do
      resources :posts, only: [:index, :show] do
        resources :comments, only: [:index, :create]
      end
    end
  end
end
```

- [ ] Adding the show action will allow us to access individual users and posts. For example

```

class Api::V1::UsersController < ApplicationController
  def index
    @users = User.all
    render json: @users
  end

  def show
    @user = User.find(params[:id])
    render json: @user
  end
end

```

#### Kindly use that approach on your posts too